obsei/preprocessor/base_text_cleaner.py

```diff
@@ -0,0 +1,51 @@
+from abc import abstractmethod
+from typing import Any, Dict, List, Optional
+from enum import Enum
+from pydantic import BaseModel, PrivateAttr
+
+
+from obsei.analyzer.base_analyzer import AnalyzerRequest
+
+
+class CleaningFunctions(str, Enum):
+    to_lower_case: str = "to_lower_case"
+    remove_white_space: str = "remove_white_space"
+    remove_punctuation: str = "remove_punctuation"
+    remove_special_characters: str = "remove_special_characters"
+    decode_unicode: str = "decode_unicode"
+    remove_date_time: str = "remove_date_time"
+    replace_domain_keywords: str = "replace_domain_keywords"
+    stem_text: str = "stem_text"
+    remove_stop_words: str = "remove_stop_words"
+
+
+class BaseTextProcessorConfig(BaseModel):
```

Comment:
What I was think is following 
`BaseTextProcessorConfig` -> `BaseTextCleanerConfig` -> Many cleaners
                                                -> `BaseTextSplitterConfig` -> ....
Similarly for `BaseTextProcessor` as well.
Mainly Data Wrangling is very time consuming for the user so idea to provide configurable and pluggable pre-processing steps which can run in parallel or serial.